src/Core/Data/BoundedDeque.hpp

```
@@ -0,0 +1,136 @@
+	/* Pops data from the front of the deque. Blocks if the deque is empty until data is available.
+		@return The popped data.
+	*/
+	T pop_front() {
+		return pop(
+			[&]() -> T& { return m_deque.front(); },
+			[&]() { m_deque.pop_front(); }
+		);
+	}
+
+
+	size_t size() const {
```

Clang (14.0.6)

[warning] function 'size' should be marked [[nodiscard]]

(modernize-use-nodiscard)